Client/Frontend/Home/HistoryHighlights/FxHomeHistoryHighlightsVC.swift

```diff
@@ -0,0 +1,239 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import Storage
+
+struct HistoryHighlightsCollectionCellConstants {
+    static let maxNumberOfItemsPerColumn = 3
+    static let maxNumberOfColumns = 3
+}
+
+class FxHomeHistoryHighlightsCollectionCell: UICollectionViewCell, ReusableCell {
+
+    // MARK: - Properties
+    var profile: Profile?
+    var viewModel: FxHomeHistoryHightlightsVM?
+
+    // MARK: - Variables
+    var numberOfColumns: Int {
+        guard let count = viewModel?.historyItems.count else { return 0 }
+        return Int(ceil(Double(count) / Double(HistoryHighlightsCollectionCellConstants.maxNumberOfItemsPerColumn)))
+    }
+
+    // MARK: - UI Elements
+    lazy var collectionView: UICollectionView = {
+        let collectionView = UICollectionView(frame: .zero,
+                                              collectionViewLayout: compositionalLayout)
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        collectionView.isScrollEnabled = true
+        collectionView.alwaysBounceVertical = false
+        collectionView.showsHorizontalScrollIndicator = false
+        collectionView.backgroundColor = UIColor.clear
+        collectionView.clipsToBounds = true
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.register(HistoryHighlightsCell.self,
+                                forCellWithReuseIdentifier: HistoryHighlightsCell.cellIdentifier)
+
+        return collectionView
+    }()
+
+    // MARK: - Inits
+    convenience init(frame: CGRect,
+                     with profile: Profile,
+                     and viewModel: FxHomeHistoryHightlightsVM) {
+        self.init(frame: frame)
+        setupLayout()
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupLayout()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Helper methods
+    private func setupLayout() {
+        contentView.addSubview(collectionView)
+
+        NSLayoutConstraint.activate([
+            collectionView.topAnchor.constraint(equalTo: contentView.topAnchor, constant: 2),
+            collectionView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            collectionView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -2),
+            collectionView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor)
+        ])
+    }
+
+    private lazy var compositionalLayout: UICollectionViewCompositionalLayout = {
+        let item = NSCollectionLayoutItem(
+            layoutSize: NSCollectionLayoutSize(widthDimension: .fractionalWidth(1),
+                                               heightDimension: .estimated(65))
```

Comment:
`65` could be from a constant and be called cellHeight or something? It's used on line 87 as well.